### PR TITLE
[dlib] Updated dlib to v19.19

### PR DESCRIPTION
**Pull request by stoperro**
Updated dlib from v19.17 to v19.19 which at the time of writing is newest official release: Dec 14, 2019 (http://dlib.net/release_notes.html)

Additional changes to port files I had to do:
- As I did minimal changes to original dlib files, now dlib may test if libpng and libjpeg works. Still port files are patching original CMake files, so that libpng and libjpeg are REQUIRED and hopefully will fail if not found/working.
- Removed patch for Mac as it failed to apply on upstream, and looked as no longer necessary (original CMakeLists.txt no longer checks for APPLE there).

So far tested only on x64-windows, namely dlib[cuda]:x64-windows.

### CODE REVIEW

File: ports/dlib/portfile.cmake [hunk 1]
```diff
@@ -5,11 +5,10 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
```
Comment (LilyWangL) [inline ports/dlib/portfile.cmake:5]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/dlib/CONTROL [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Source: dlib
-Version: 19.17-1
+Version: 19.19-1
```
Comment (LilyWangL) [inline ports/dlib/CONTROL:2]: This version should be 19.19.

File: ports/dlib/portfile.cmake [hunk 3]
```diff
@@ -67,6 +66,8 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_cu
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_cpp11)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_avx)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_sse4)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_libjpeg)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_libpng)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/external/libpng/arm)
 
 # Dlib encodes debug/release in its config.h. Patch it to respond to the NDEBUG macro instead.
```
Comment (LilyWangL) [inline ports/dlib/portfile.cmake:73]: Can you please use command `file(INSTALL ${SOURCE_PATH}/dlib/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by stoperro** [comment]
Hmm, the CI fail is in "osg" port, which I suspect doesn't depend on dlib. Is it some flacky test situation?

**Review by LilyWangL — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the update!

**Pull request merged**